### ISSUE TITLE
Rule `jsdoc/require-returns` doesn’t error on tag `@returns` missing for an async function that returns something

### ISSUE THREAD

**Issue by jaydenseric** [gajus/eslint-plugin-jsdoc#1019]
## Expected behavior

The rule `jsdoc/require-returns` should require a `@returns` tag in a JSDoc comment for an async function that explicitly returns something.

## Actual behavior

The rule `jsdoc/require-returns` never requires a `@returns` tag in a JSDoc comment on an async function that explicitly returns something.

## ESLint Config

```js
{
  plugins: ["jsdoc"],
  rules: {
    "jsdoc/require-returns": "error"
  }
}
```

## ESLint sample

```js
/**
 * Description.
 */
export default async function demo() {
  return true;
}
```

## Environment

- Node version: v19.9.0
- ESLint version v8.38.0
- `eslint-plugin-jsdoc` version: v41.1.1

**Comment by brettz9** [comment]
This is by design. You can set the `forceReturnsWithAsync` option to `true` to report cases as you describe. The rationale for the default behavior is that `async` itself flags that the return type will be a `Promise`, but it is indeed stricter and perhaps a good candidate for a `strict` config. Closing but feel free to comment further as needed.

**Comment by jaydenseric** [comment]
I already understood the `forceReturnsWithAsync` option. It forces you to require a `@return` tag for all async functions, even one with implicit return of a promise. For example:

```js
/**
 * Description.
 *
 * @returns {Promise<void>}
 */
export default async function demo() {}
```

This is not desirable.

The documented default behavior for the rule `require-returns` is that it should still require a tag `returns` if the async function ***explicitly*** returns a value, which is desired:

> - `forceReturnsWithAsync` - By default `async` functions that do not explicitly return a value pass this rule as an `async` function will always return a `Promise`, even if the `Promise` resolves to void. You can force all `async` functions (including ones with an explicit `Promise` but no detected non-`undefined` `resolve` value) to require `@return` documentation by setting `forceReturnsWithAsync` to `true` on the options object. This may be useful for flagging that there has been consideration of return type. Defaults to `false`.

Unfortunately the documented behavior is not what happens; hence this bug report.

**Comment by brettz9** [comment]
Very sorry about that. I guess I had it in mind that we might need an `ignoreAsync` to the rule for those who think the `async` is enough to indicate the return type (I don't think so, but I could see some projects not caring in that case but still wanting the return type otherwise). Perhaps that was the cause when I introduced a regression as well.

**Comment by jaydenseric** [comment]
Thanks @brettz9 for the speedy fix! I highlighted in a team meeting today what a great job you do with this project.
